### DISCUSSION TITLE
Documentation

### DISCUSSION THREAD

**Discussion by WoistdasNiveau** [Rahix/avr-hal#434] [Q&A]
Dear Community!

Is there some kind of documentation of all methods and macros available etc? Currently i am stuck with finding the equivalent to the digitalWrite and analogWrite methods. Some kind of overview of all methods would be nice.

**Comment by Rahix** [comment]
All we have at the moment is

1. Online API reference: https://rahix.github.io/avr-hal/arduino_hal/index.html
2. A lot of examples: https://github.com/Rahix/avr-hal/tree/main/examples/arduino-uno/src/bin

**Comment by Galz648** [comment]
to do analog write you need to use pwm (pulse width modulation). refer to the documentation for this. @WoistdasNiveau. feel free to contact me if you need help
